### PR TITLE
- Transaction detail screen has been completed.

### PR DESCRIPTION
**Pull request by yasirinitialcode**
## Related Issue
- Designing transaction detail activity.

ISSUE LINK: https://app.asana.com/0/1203144541549209/1204097456360678

## Description
- Transaction detail activity was designed.
- Views were customized according to 3 type of transaction.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?
- Tested with Redmi Note 7

## Screenshots or Sample Output

https://user-images.githubusercontent.com/115461849/224752956-245f5acd-25ae-4c69-bb84-824960274902.mp4

### CODE REVIEW

File: app/src/main/java/com/netmera/netmerafintech/ui/home/TransactionDetailActivity.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        binding.apply {
+            averageSpentAmount.text = transaction.price
+            averageSpentAmount.setTextColor(ContextCompat.getColor(this@TransactionDetailActivity, transaction.priceColor))
+            totalSpentAmount.setTextColor(ContextCompat.getColor(this@TransactionDetailActivity, transaction.priceColor))
+            transactionIcon.setImageResource(transaction.icon)
+            transactionIconBackground.setCardBackgroundColor(ContextCompat.getColor(this@TransactionDetailActivity, transaction.iconContainerColor))
+            transactionName.text = transaction.name
+            transactionType.text = transaction.type
+
+            // custom settigns, no need for id == 0, it's set already
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/ui/home/TransactionDetailActivity.kt]: You should also add these to transaction object.

File: app/src/main/java/com/netmera/netmerafintech/utils/Extensions.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.netmera.netmerafintech.utils
+
+import android.app.Activity
+import android.widget.Toast
+
+fun Activity.toast(message: String) {
+    Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/utils/Extensions.kt:14]: Nice!

File: app/src/main/java/com/netmera/netmerafintech/ui/home/TransactionDetailActivity.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        }
+        transaction?.let {
+            initViews(it)
+        } ?: run {
+            toast("Error occurred while opening transaction detail. Please try again.")
+            finish()
+            return
+        }
+
+        setOnClickActions()
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/ui/home/TransactionDetailActivity.kt]: You should move this after `initViews(it)` line

**Review by ghost — APPROVED** [review]

File: app/src/main/java/com/netmera/netmerafintech/utils/Extensions.kt [hunk 4]
```diff
@@ -1,8 +1,17 @@
...
 
 fun Activity.toast(message: String) {
     Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
+}
+
+// getParcelable needs to be set class of data type after API level 33
+inline fun <reified T : Parcelable> Bundle.parcelable(key: String): T? = when {
+    SDK_INT >= 33 -> getParcelable(key, T::class.java)
+    else -> @Suppress("DEPRECATION") getParcelable(key) as? T
 }
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/utils/Extensions.kt:21]: Nice!

**Review by ghost — APPROVED** [review]

**Review by bsoykal — APPROVED** [review]
very good, thanks

**Pull request merged**